### PR TITLE
Fix for bug issue OpenRCT2#8219

### PR DESCRIPTION
**Pull request by hokasha2016**
Fixed issue that a new folder would be created even though it was deleted by the user. Code now checks for path existence first, if the path exists (folder wasn't deleted) then it opens that folder. If the folder was deleted, then it takes you to the default path.

https://github.com/OpenRCT2/OpenRCT2/issues/8219

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Could you remove the formatting changes so that you are only touching the files that changed. `contributors.md`, `src/openrct2-ui/windows/LoadSave.cpp`. If you could also clean up the commit history that would be great but if your not sure then can squash merge it to make it sensible.

**Comment by IntelOrca** [comment]
To clean up history (remove merge commits) you need to rebase against the latest version of develop.

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 1]
```diff
@@ -237,20 +238,22 @@ static const char* getFilterPatternByType(const int32_t type, const bool isSave)
     return "";
 }
 
+//Checks for path existance
+bool IsPathExist(const std::string &s)
+{
+    struct stat buffer;
+    return (stat (s.c_str(), &buffer) == 0);
```
Comment (IntelOrca) [inline src/openrct2-ui/windows/LoadSave.cpp]: `stat` is not available for every platform. You instead use `platform_directory_exists` which is already implemented correctly for each platform.

**Comment by hokasha2016** [comment]
hey im trying to rebuild and clean the commits, could you help me?

**Comment by duncanspumpkin** [comment]
@hokasha2016 its slightly harder with you having worked directly on develop but its still possible. First make a branch that will save all your work. `git checkout -b fix_8974` then go back to develop `git checkout develop`. Then you need to reset develop back upstream. I'm unsure what you have called upstream but it is probably `upstream` so you would do `git reset --hard upstream/develop`. At this point develop will match OpenRCT2/develop. Then you can go to your feature branch `git checkout fix_8974` and rebase it interactively `git rebase develop -i`. When this comes up change every line to a `squash` or `s` instead of a `pick` `p`. Now you have a nice simple single commit which has your change. Now you can push that to github. The problem now is that this PR is against your develop. So to prevent closing that and reopening it you now need to go back to develop `git checkout develop` then merge in your feature `git merge fix_8974` and finally `git push origin develop --force`.

**Comment by bdupree5** [comment]
We followed your steps and it still didn't seem to work. We have someone we can ask about it tomorrow and we will let you know what happens.

**Comment by duncanspumpkin** [comment]
Where did you get into trouble? I didn't actually test my commands myself just doing it from memory so easy for it to be wrong.

**Comment by bdupree5** [comment]
@duncanspumpkin  I do have a question, doing the steps, will we see the changes in github? As up above we can see all the commits we made. Will the re-base delete or squash the previous commits and show it on the website? We've tried reset as well. Will the changes show on github?

**Comment by duncanspumpkin** [comment]
Yes the idea would be that it would show up on this PR. From looking at your github setup i can see you now have your develop back to normal with none of your commits. Since this is now closed I would start from scratch make a new branch `git checkout -b fix_8219` then make your changes then commit them to the branch then `git push fix_8219 origin` or whatever your github remote is. Then you can PR with just that one branch. For reference if you type in `git remote -v` you can see all of your remote names and then I could customise it for what you have it setup as.

**Pull request closed without merging**